src/java.base/share/classes/java/nio/file/CopyMoveHelper.java

```diff
@@ -137,10 +137,14 @@ static void copyToForeignTarget(Path source, Path target,
         if (sourceAttrs.isSymbolicLink())
             throw new IOException("Copying of symbolic links not supported");
 
+        // delete target if it exists and REPLACE_EXISTING is specified
+        if (opts.replaceExisting) {
+            Files.deleteIfExists(target);
+        } else if (Files.exists(target))
+            throw new FileAlreadyExistsException(target.toString());
+
         // create directory or copy file
         if (sourceAttrs.isDirectory()) {
-            if (opts.replaceExisting)
-                Files.deleteIfExists(target);
             Files.createDirectory(target);
         } else {
             try (InputStream in = Files.newInputStream(source)) {
```

Comment:
If we are you reverting that then I assume you also should revert the change to L147 too, meaning replaceExistingOrEmpty() isn't needed now.

Reply:
`replaceExistingOrEmpty()` removed in 4a34ff3b36bbcbd43139f7888ac4d953ecef91da.